File: 27. LeetCode Problems/1578. Minimum_Time_to_Make_Rope_Colorful.cpp
class Solution {
public:
    int minCost(string colors, vector<int>& neededTime) {
        int ans = 0;
        for(int i = 0; i < colors.size() - 1; i++) {
            if(colors[i] == colors[i+1]) {
                if(neededTime[i] > neededTime[i+1]) {
                    ans += neededTime[i+1];
                    neededTime[i+1] = neededTime[i];
                } else
                    ans += neededTime[i];
            }
        }
        return ans;
    }
};

File: 27. LeetCode Problems/1578_Minimum Time to Make Rope Colorful.py
'''
https://leetcode.com/problems/minimum-time-to-make-rope-colorful/
'''
class Solution:
    def minCost(self, colors: str, neededTime: List[int]) -> int:
        total_time = 0
        i, j = 0, 0
        
        while i < len(neededTime) and j < len(neededTime):
            curr_total = 0
            curr_max = 0
            
            while j < len(neededTime) and colors[i] == colors[j]:
                curr_total += neededTime[j]
                curr_max = max(curr_max, neededTime[j])
                j += 1

            total_time += curr_total - curr_max
            i = j
        
        return total_time
